prefect_docker/credentials.py

```diff
@@ -0,0 +1,49 @@
+"""Module containing docker credentials."""
+import docker
+from prefect import get_run_logger
+from prefect.blocks.core import Block
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+from pydantic import Field, SecretStr
+
+
+class DockerRegistryCredentials(Block):
+    """
+    Block used to manage credentials for interacting with a Docker Registry.
+    """
+
+    _block_type_name = "Docker Registry Credentials"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/2IfXXfMq66mrzJBDFFCHTp/6d8f320d9e4fc4393f045673d61ab612/Moby-logo.png?h=250"  # noqa
+    _description = "Store credentials for interacting with a Docker Registry."
+
+    username: str = Field(
+        default=..., description="The username to log into the registry with."
+    )
+    password: SecretStr = Field(
+        default=..., description="The password to log into the registry with."
+    )
+    registry_url: str = Field(
+        default=...,
+        description=(
+            'The URL to the registry. Generally, "http" or "https" can be omitted.'
+        ),
```

Comment:
This example is useful if you're using Docker Hub.
```suggestion
        default=...,
        description=(
            'The URL to the registry. Generally, "http" or "https" can be omitted.'
        ),
        example="registry.hub.docker.com",
```